### PR TITLE
Book Suggestion Model

### PR DESCRIPTION
**Pull request by cquiro**
## Summary

- Create BookSuggestion model using rails generator
- Run migrations
- Add model validations
- Create BookSuggestion factory
- Test validations

## Trello Card

https://trello.com/c/p1PuRA7r/28-crear-modelo-de-booksuggestion

### CODE REVIEW

File: db/migrate/20181001191550_create_book_suggestions.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  def change
+    create_table :book_suggestions do |t|
+      t.string :synopsis, default: ''
+      t.float :price, default: 0.0
+      t.string :author, null: false, default: ''
+      t.string :title, null: false, default: ''
+      t.string :link, null: false, default: ''
+      t.string :publisher, null: false, default: ''
+      t.string :year, null: false, default: ''
+      t.references :user, foreign_key: true
```
Comment (facucompe) [inline db/migrate/20181001191550_create_book_suggestions.rb:11]: We could add an idex for this reference

File: app/models/book_suggestion.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class BookSuggestion < ApplicationRecord
+  validates :author, :title, :link, :publisher, :year, presence: true
+
+  belongs_to :user
```
Comment (facucompe) [inline app/models/book_suggestion.rb:4]: What about the user? Should he be able to access to his books suggestions

**Review by facucompe — CHANGES_REQUESTED** [review]

File: db/migrate/20181001191550_create_book_suggestions.rb (hunk 1 above)
Comment (cquiro) [reply db/migrate/20181001191550_create_book_suggestions.rb:11]: @facucompe it is indexed 👍 

In Rails 5, the index option defaults to true for add_reference:

https://api.rubyonrails.org/v5.2/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_reference

File: app/models/book_suggestion.rb (hunk 2 above)
Comment (cquiro) [reply app/models/book_suggestion.rb:4]: @facucompe done! Added has_many :book_suggestions to User model.

**Review by facucompe — APPROVED** [review]

**Review by lmasello — APPROVED** [review]

**Pull request merged**